### ISSUE TITLE
Regent reverses the order of do...end blocks

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#406]
The following snipped prints 2 before 1:

```
import 'regent'
local task main()
  do
    regentlib.c.printf('1\n')
  end
  do
    regentlib.c.printf('2\n')
  end
end
regentlib.start(main)
```

**Comment by lightsighter** [comment]
Time for monads in Regent now @elliottslaughter? ;)

**Comment by elliottslaughter** [comment]
As a workaround, try running with `-fflow 0`.

**Comment by elliottslaughter** [comment]
Should be fixed in fbb6593b14aa0fac108504e14583e4f85418b41a.

**Comment by manopapad** [comment]
Fixed!
